src/shared/components/download/DownloadPreview.tsx

```diff
@@ -40,9 +42,13 @@ const DownloadPreview = ({
   return (
     <div className={styles.preview} ref={scrollRef}>
       <h4>Preview</h4>
-      <CodeBlock lightMode data-testid="download-preview">
-        {typeof data === 'string' ? data : JSON.stringify(data, null, 2)}
-      </CodeBlock>
+      {data ? (
+        <CodeBlock lightMode data-testid="download-preview">
+          {typeof data === 'string' ? data : JSON.stringify(data, null, 2)}
+        </CodeBlock>
+      ) : (
+        'No preview available for this format'
```

Comment:
Similar approach will be needed for the API URLs.

Reply:
Good catch, I'd forgotten to handle this case.
I've now handled it by disabling the whole API section in this case